### ISSUE TITLE
drush make silently fails when it encounters an error 

### ISSUE THREAD

**Issue by raphaeltraviss** [drush-ops/drush#412]
Consider the make file:
```
core = 7.x

api = 2

projects[] = "drupal"

; Modules

projects[] = "admin_menu"

projects[] = "advanced_help"

projects[] = "auto_entitylabel"

projects[my_module][download][type] = "git"
projects[my_module][download][url] = "ssh://git@gitlab.com/my_username/my_module.git"
projects[my_module][download][branch] = "non-existent branch"
projects[my_module][type] = "module"

projects[] = "another project"

projects[] = "yet another project"
```

When you run this makefile with drush (version 6.20) make, it starts downloading the needed projects.  When it gets to the custom module, it fails--this could be for a variety of reasons: branch doesn't exist, login credentials are bad, url is mistyped.  You will see [failed] appear in the command line output.

However, drush will continue on, happily downloading the rest of the projects.  Then, when it finishes, you are left with nothing in your target directory.

The desired operation should be:

When an error is encountered...

1. Stop drush immediately

or 

2. Download the rest of the modules that did work, and place them into the target directory

Right now, when I run this makefile, I get no fatal error messages, lots of projects downloaded, and then absolutely none of these downloaded projects added to the target directory.

**Comment by ergonlogic** [comment]
Drush's normal error reporting behaviour doesn't re-print all errors at the end of a command's execution. I don't see any compelling reason to make an exception for `make`. On the other hand, I do think an error at the end of the `drush make` command would be appropriate, just stating that the build failed.

**Comment by jnous** [comment]
At least non-zero exit code would be great, so that automated build scripts would be able to detect the failure.

**Comment by ergonlogic** [comment]
Drush make already returns a non-zero exit code on such errors.

**Comment by drupalninja** [comment]
+! on making the build fail.

**Comment by jhedstrom** [comment]
Because of the concurrent download processes, it is difficult to stop all of them since they are not aware of one another.

**Comment by ergonlogic** [comment]
Per my earlier comment, I've added an error message at the end of `drush make` should any of the projects fail to download.

Behaviours that have not changed, but address some of the concerns brought up in this issue:
* `drush make` still returns a proper exit code: `0` on success, `1` on failure.
* If a download error occurs when the `--force-complete` flag is passed, only warnings are emitted, thus bypassing the final error, and the process will return `0`.

A change in behaviour that I've noted in the master branch, un-related to my recent commit, is that the codebase will build regardless of errors during download (w/o `--force-complete`). I believe this to be a bug, and I'll continue to investigate it in #1367.

**Comment by ergonlogic** [comment]
Actually, it looks like a06cde9 fixed #1367 too. So disregard the final paragraph in my previous comment.
